test/unit/query/builder.js

```diff
@@ -743,6 +743,45 @@ describe('QueryBuilder', () => {
     );
   });
 
+  it('clear by statements', () => {
```

Comment:
More tests needed to make sure that each clear statement works separately and doesn't clear anything unexpected + test for non-happy-path while using non existing statement.  

Also some test showing that it does not clear anything from subquery would be good to have.

Reply:
I agree, I'll do it.